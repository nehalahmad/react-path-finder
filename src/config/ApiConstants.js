// config/ApiConstants.js
/**
 * API status
 */
export const SUCCESS = "success";
export const IN_PROGRESS = "in progress";
export const FAIL = "failure";
/**
 * some other constants
 */
export const NUMBER_ATTEMPTS = 3;
