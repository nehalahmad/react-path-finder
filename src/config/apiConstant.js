
/**
 * google api key
 */
export const googleAPIKey = 'AIzaSyBG0SybP0EKWH3Jvwki7IR5AMyO_cUeeQc'
/**
 * Base url of back end
 */
export const BASE_URL = "https://mock-api.dev.lalamove.com/";
export const ROUTE_API = 'route';

/**
 * API status
 */
export const SUCCESS = 'success';
export const IN_PROGRESS = 'in progress';
export const FAIL = 'failure';
export const NUMBER_ATTEMPTS = 3;