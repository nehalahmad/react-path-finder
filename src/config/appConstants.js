export const SUBMIT_TEXT = "Submit";
export const RESUBMIT_TEXT = "Re-Submit";
export const RESET_TEXT = "Reset";
export const DIRECTION_NOT_FOUND = "Direction couldn't found!";
export const SOMETHING_WRONG = "Oops, something went wrong!";
export const LOCATION = {
  ONE: {
    TITLE: "Starting location",
    PLACEHOLDER: "enter starting location"
  },
  TWO: {
    TITLE: "Drop-off point",
    PLACEHOLDER: "enter drop-off point"
  }
};
export const ERROR_DIRECTION_SERVICE = "Error in direction service response";
export const UNAUTHORISED_LOCATIONS = "Locations couldn't authorized!";
export const TOTAL_DISTANCE = "Total distance:";
export const TOTAL_TIME = "Total time:";