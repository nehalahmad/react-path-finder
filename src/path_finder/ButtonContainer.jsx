import React from "react";
import PropTypes from "prop-types";
import { Button } from "react-bootstrap";
import { RESET_TEXT } from "../config/AppConstants";

/**
 * @description: A wrapper container for form buttons
 */
const ButtonContainer = props => {
  const { isLoader, submitBtnText } = props;

  return (
    <div className="button-container text-right">
      <Button type="submit" variant="primary" disabled={isLoader}>
        {submitBtnText}
      </Button>
      <Button type="reset" variant="reset">
        {RESET_TEXT}
      </Button>
    </div>
  );
};

// validate prop types
ButtonContainer.propTypes = {
  isLoader: PropTypes.bool,
  submitBtnText: PropTypes.string
};

export default ButtonContainer;
