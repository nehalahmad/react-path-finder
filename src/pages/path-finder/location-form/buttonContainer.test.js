import React from "react";
import ButtonContainer from "./buttonContainer";
import { shallow } from "enzyme";
import { SUBMIT_TEXT, RESUBMIT_TEXT, RESET_TEXT } from "../../../config/appConstants";

describe("<ButtonContainer />", () => {
  it("submit button renders", () => {
    const buttonContainerObj = shallow(
      <ButtonContainer submitBtnText={SUBMIT_TEXT} />
    );
    expect(buttonContainerObj).toMatchSnapshot();

    let mockCallBack = jest.fn();

    const btnSubmit = buttonContainerObj.find("[type='submit']");
    if (btnSubmit) {
      expect(btnSubmit.text()).toEqual(SUBMIT_TEXT);
      btnSubmit.simulate("click");
      expect(mockCallBack.mock.calls.length).toEqual(0);
    }

    const btnReset = buttonContainerObj.find("[type='reset']");
    if (btnReset) {
      expect(btnSubmit.text()).toEqual(RESET_TEXT);
      btnReset.simulate("click");
      expect(mockCallBack.mock.calls.length).toEqual(0);
    }
  });

  it("submit button renders", () => {
    const buttonContainerObj = shallow(
      <ButtonContainer submitBtnText={RESUBMIT_TEXT} />
    );
    expect(buttonContainerObj).toMatchSnapshot();
    const btnSubmit = buttonContainerObj.find("[type='submit']");
    if (btnSubmit) {
      expect(btnSubmit.text()).toEqual(RESUBMIT_TEXT);

      let mockCallBack = jest.fn();
      btnSubmit.simulate("click");
      expect(mockCallBack.mock.calls.length).toEqual(0);
    }
  });
});
