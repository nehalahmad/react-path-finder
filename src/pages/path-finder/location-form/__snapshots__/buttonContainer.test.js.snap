// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonContainer /> reset button renders 1`] = `ShallowWrapper {}`;

exports[`<ButtonContainer /> submit button renders 1`] = `ShallowWrapper {}`;

exports[`<ButtonContainer /> submit button renders 2`] = `ShallowWrapper {}`;

exports[`<FormButton /> reset button renders 1`] = `ShallowWrapper {}`;

exports[`<FormButton /> submit button renders 1`] = `ShallowWrapper {}`;
